### PR TITLE
Add Makefile

### PR DESCRIPTION
**Pull request by imbsky**

### CODE REVIEW

**Comment by imbsky** [comment]
Hmm? Somehow the permissions of some files seem to have been wrong before and that change is included in the commit. Should I divide it into separate commits?

**Comment by prometheansacrifice** [comment]
I think it's fine. Regarding the wrapping parens, I'll leave it up to you
all. I'm fine either way.

File: package.json [hunk 1]
```diff
@@ -280,9 +280,7 @@
     "vscode:prepublish": "yarn run build",
     "package": "vsce package --yarn",
     "fmt:check": "prettier --check .",
-    "fmt:ocaml": "git ls-files '*.ml' '*.mli' | xargs ocamlformat -i",
```
Comment (rgrinberg) [inline package.json:283]: Why did you replace `git ls-files` with find?

Comment (imbsky) [reply package.json:283]: I don't know, but it seems to make more sense to use `git ls-files'. I will update this.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,12 @@
+OCAML_SRCFILES = $(shell git ls-files "*.ml" "*.mli")
+REASON_SRCFILES = $(shell git ls-files "*.re" "*.rei")
```
Comment (rgrinberg) [inline Makefile:2]: Why do we have formatting of reason sources? We eliminated all reason sources already.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+build:
+	yarn build
+.PHONY: build
+
+fmt:
+	esy ocamlformat --inplace --enable-outside-detected-project $(OCAML_SRCFILES)
+	esy refmt --in-place $(REASON_SRCFILES)
```
Comment (rgrinberg) [inline Makefile:10]: Actually, I don't use esy so this is making formatting unusable for me. If someone wants to use ocamlformat from esy, they should just run `$ esy make fmt`.

File: Makefile (hunk 2 above)
Comment (imbsky) [reply Makefile:2]: There are several Reason codes in the fixtures directory. Well, the interface file doesn't exactly exist yet.

File: Makefile (hunk 3 above)
Comment (imbsky) [reply Makefile]: Updated.

File: Makefile (hunk 2 above)
Comment (rgrinberg) [reply Makefile:2]: That's a good opportunity to convert them to OCaml then.

Comment (prometheansacrifice) [reply Makefile:2]: Fixture are for e2e test. We need to ensure the plugin works with Reason syntax

**Comment by rgrinberg** [comment]
In that case, the script should use the refmt that comes with bucklescript.

**Comment by rgrinberg** [comment]
Could you add `doc-comments-val=after` to the ocamlformat config so the comment style doesn't change?

**Comment by imbsky** [comment]
The `doc-comments-val` option was added in 0.14.0, so I don't think we can use it.

**Comment by rgrinberg** [comment]
I might be missing something - why did you downgrade to 0.13.0?

**Comment by imbsky** [comment]
The minimum version of dune was changed from 0.14.0, but esy can't handle it without a dirty hack. and I didn't have a strong motivation to update to 0.14.0 or above, so I reverted it.

**Comment by rgrinberg** [comment]
Okay. Again, it would have been easier to understand this from the description :)

**Comment by imbsky** [comment]
The version of ocamlformat was downgraded by another PR. But I agree with that.

**Comment by imbsky** [comment]
I use opam for all OCaml projects, so if we need to use 0.14.1 in this project, I don't need any dirty hacks. Is there any reason why we are using esy in this project?

**Comment by rgrinberg** [comment]
We are only using esy to test that using the extension is working with esy. We don't need esy to develop the extension itself. If there are problems installing the appropriate ocamlformat with esy, that should be solved in esy.

**Comment by imbsky** [comment]
Oh, I see.

I don't know how to use a secondary compiler with esy, could you help me? @rusty-key @prometheansacrifice

**Comment by imbsky** [comment]
Rebased to current master. The purpose of this PR is to add a Makefile, so I would like to upgrade ocamlformat in another PR. What do you think? @rgrinberg

**Comment by rgrinberg** [comment]
Okay. We can ask one of our esy experts to update to fix the ocamlformat + esy situation

**Comment by prometheansacrifice** [comment]
Sure!

You just need the resolutions mentioned at 
https://github.com/esy/esy/issues/1062#issuecomment-601378849

Secondary compilers, unlike as in opam, are installed separately in the libroot. So, findlib needs to be told of this location. Those overrides do just that.

**Comment by imbsky** [comment]
@prometheansacrifice Thank you so much!

**Pull request merged**